### PR TITLE
Add freetype support for harfbuzz.

### PR DESCRIPTION
**Pull request by vlj**
This adds support for freetype in harfbuzz.

### CODE REVIEW

File: ports/harfbuzz/portfile.cmake [hunk 1]
```diff
@@ -18,22 +18,30 @@ vcpkg_download_distfile(ARCHIVE
...
+    SOURCE_PATH ${SOURCE_PATH}
+    PATCHES "${CMAKE_CURRENT_LIST_DIR}/0001-Add-an-extra-path-input-for-Freetype.patch"
+)
+
+file(TO_NATIVE_PATH "${VCPKG_ROOT_DIR}/installed/${TARGET_TRIPLET}/include" FREETYPE_INCLUDE_DIR)
+file(TO_NATIVE_PATH "${VCPKG_ROOT_DIR}/installed/${TARGET_TRIPLET}/lib" FREETYPE_LIB_DIR_REL)
+
 vcpkg_execute_required_process(
-    COMMAND ${NMAKE} -f Makefile.vc CFG=debug
+    COMMAND ${NMAKE} -f Makefile.vc CFG=debug FREETYPE=1 FREETYPE_DIR=${FREETYPE_INCLUDE_DIR} VCPKG_LIB_DIR=${FREETYPE_LIB_DIR_REL}
```
Comment (ras0219-msft) [inline ports/harfbuzz/portfile.cmake]: I think this should use the debug lib, not the release lib.

**Comment by vlj** [comment]
Fixed the paths for the debug build.

**Comment by ras0219-msft** [comment]
Thanks for that fix, it looks like it was more detailed than just adding an extra CMake parameter 👍.

I also corrected the casing of `Build-depends: Freetype` to be `Build-Depends: freetype` -- in the future we need to check this in vcpkg automatically.

**Comment by ebraminio** [comment]
This was my concern for some days. Thank you @vlj :)

**Comment by ebraminio** [comment]
Also feel free to upstream your patches to harfbuzz, I can also help you there :)

**Comment by ebraminio** [comment]
@vlj have a look [at this](https://github.com/behdad/harfbuzz/pull/349) please.

**Pull request merged**